core/src/main/java/org/elasticsearch/common/util/CancellableThreads.java

```diff
@@ -64,6 +65,7 @@ protected void onCancel(String reason, @Nullable Exception suppressedException)
         throw e;
     }
 
+    @SuppressForbidden(reason = "We really want to use Thread.interrupted() here")
```

Comment:
can we something more descriptive as to why? it's obvious for those who knows the class, but for those who don't this doesn't help :)